moveit_ros/visualization/motion_planning_rviz_plugin/src/ui/motion_planning_rviz_plugin_frame.ui

```diff
@@ -704,59 +756,149 @@
             <item>
-             <widget class="QCheckBox" name="allow_external_program">
-              <property name="text">
-               <string>Allow External Comm.</string>
+             <widget class="QCheckBox" name="approximate_ik">
+              <property name="sizePolicy">
+               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+                <horstretch>0</horstretch>
+                <verstretch>0</verstretch>
+               </sizepolicy>
               </property>
-              <property name="checked">
-               <bool>false</bool>
+              <property name="minimumSize">
+               <size>
+                <width>30</width>
+                <height>0</height>
+               </size>
+              </property>
+              <property name="toolTip">
+               <string>Allow approximate IK solutions</string>
+              </property>
+              <property name="text">
+               <string>Approx IK Solutions</string>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QCheckBox" name="use_cartesian_path">
+             <widget class="QCheckBox" name="allow_replanning">
+              <property name="sizePolicy">
+               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+                <horstretch>0</horstretch>
+                <verstretch>0</verstretch>
+               </sizepolicy>
+              </property>
+              <property name="minimumSize">
+               <size>
+                <width>30</width>
+                <height>0</height>
+               </size>
+              </property>
+              <property name="toolTip">
+               <string>Allows the robot to replan if the plan to be executed collides with the environment.</string>
+              </property>
               <property name="text">
-               <string>Use Cartesian Path</string>
+               <string>Replanning</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
-             <widget class="QCheckBox" name="collision_aware_ik">
+             <widget class="QCheckBox" name="allow_looking">
+              <property name="sizePolicy">
+               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+                <horstretch>0</horstretch>
+                <verstretch>0</verstretch>
+               </sizepolicy>
+              </property>
+              <property name="minimumSize">
+               <size>
+                <width>30</width>
+                <height>0</height>
+               </size>
+              </property>
+              <property name="toolTip">
+               <string>Allows the robot to reposition its sensor to look around. A sensor needs to be set up and the planning algorithm has to support this, otherwise it has no effect.</string>
```

Comment:
Again, referring to a tutorial might be helpful here.

Reply:
```suggestion
               <string>Allow to reposition a sensor to resolve apparent collisions. A sensor needs to be set up and the new measurements have to affect the collision scene, otherwise it has no effect.</string>
```

Reply:
I never used this feature and it is completely unclear to me, which strategy is used to reposition the sensor.

Reply:
The logic is quite straight-forward: If the cost of obstacles close to the path is too high, direct the sensor towards a cost source. It was used to get rid of kinect-based artefact in the octomap in the past and is implemented [here](https://github.com/ros-planning/moveit/blob/master/moveit_ros/planning/plan_execution/src/plan_with_sensing.cpp#L206).